File: app/src/main/java/br/pedroso/upcomingmovies/movieslist/adapter/MovieViewHolder.java
package br.pedroso.upcomingmovies.movieslist.adapter;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.recyclerview.widget.RecyclerView;

import com.squareup.picasso.Picasso;

import br.pedroso.upcomingmovies.databinding.ItemMovieBinding;
import br.pedroso.upcomingmovies.domain.Movie;

public class MovieViewHolder extends RecyclerView.ViewHolder {

    private OnMovieClickListener onMovieClickListener;

    private ItemMovieBinding binding;

    private MovieViewHolder(ItemMovieBinding binding, OnMovieClickListener onMovieClickListener) {
        super(binding.getRoot());
        this.binding = binding;
        this.onMovieClickListener = onMovieClickListener;
    }

    public void bind(Movie movie) {
        binding.textViewMovieTitle.setText(movie.getTitle());

        // TODO: deal with null images
        Picasso.with(binding.getRoot().getContext()).load(movie.getPosterPath()).into(binding.imageViewMoviePoster);

        binding.textViewMovieReleaseDate.setText(movie.getReleaseDate());

        binding.getRoot().setOnClickListener(v -> {
            if (onMovieClickListener != null) {
                onMovieClickListener.onMovieClick(movie);
            }
        });
    }

    public static MovieViewHolder create(ViewGroup parent, OnMovieClickListener onMovieClickListener) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());

        ItemMovieBinding binding = ItemMovieBinding.inflate(inflater, parent, false);

        return new MovieViewHolder(binding, onMovieClickListener);
    }
}

File: app/src/main/kotlin/br/pedroso/upcomingmovies/movieslist/adapter/MovieViewHolder.kt
package br.pedroso.upcomingmovies.movieslist.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import br.pedroso.upcomingmovies.databinding.ItemMovieBinding
import br.pedroso.upcomingmovies.domain.Movie
import com.squareup.picasso.Picasso

class MovieViewHolder private constructor(
    private val binding: ItemMovieBinding,
    private val onMovieClickListener: OnMovieClickListener
) : RecyclerView.ViewHolder(binding.root) {

    fun bind(movie: Movie) {
        binding.textViewMovieTitle.text = movie.title

        // TODO: deal with null images
        Picasso.with(binding.root.context).load(movie.posterPath).into(binding.imageViewMoviePoster)

        binding.textViewMovieReleaseDate.text = movie.releaseDate

        binding.root.setOnClickListener {
            onMovieClickListener.onMovieClick(movie)
        }
    }

    companion object {
        @JvmStatic
        fun create(parent: ViewGroup, onMovieClickListener: OnMovieClickListener): MovieViewHolder {
            val inflater = LayoutInflater.from(parent.context)

            val binding = ItemMovieBinding.inflate(inflater, parent, false)

            return MovieViewHolder(binding, onMovieClickListener)
        }
    }
}
